src/app/forms/components/date/date.component.ts

```diff
@@ -158,7 +157,22 @@ export class DateComponent extends BaseControlComponent implements OnInit, OnDes
     this.control?.addValidators([DateValidators.validDate(this.displayTime, this.label)]);
   }
 
-  validate() {
+  validate(event: Event) {
+    this.addValidators();
+    if (!this.day) {
+      return;
+    }
+
+    if (!this.month) {
+      return;
+    }
+    if (!this.year) {
+      return;
+    }
+
+    if (this.year && event.type !== 'blur') {
+      return;
+    }
     this.errors = validateDate(this.day || '', this.month || '', this.year || '', this.label);
   }
 
```

Comment:
I would suggest handling the displaying or not of errors in the `elementHasErrors` method below. We ideally don't want to stop validators running or anything like that.

Additionally you could toggle the visibility of the `<app-field-error-message` component in the `html` of the component using some logic based in the `ts` of the component. Something along the lines of all the elements being touched for instance.

Reply:
It seems like the function elementHasErrors doesn't have access to the event, not sure how I could access the event type otherwise? As I can't pass in the event into it

elementHasErrors(i: number, event: Event) --> produces an error in the html